### ISSUE TITLE
gitea_repository is not declarative when "username" is a non-existent user or organization

### ISSUE THREAD

**Issue by juddowilcox** [Lerentis/terraform-provider-gitea#14]
**Describe the bug**
When create gitea_repository resource and "username" is a non-existent organization (or user) then the result does not match the declaration.

**To Reproduce**
```
terraform {
  required_providers {
    gitea = {
      source = "lerentis/gitea"
      version = "0.11.0"
    }
  }
}
provider "gitea" {
  base_url = var.gitea_url
  token    = var.gitea_token
}
variable "gitea_url" {
}
variable "gitea_token" {
}
#
# well-controlled dependency, correct result
# result matches declaration after one apply operation
#
resource "gitea_org" "repotest0" {
  name = "RepoTest0"
}
resource "gitea_repository" "repotest0" {
  username = gitea_org.repotest0.name
  name = "repotest0"
}
output repotest0 {
  value = "declared=RepoTest0 repotest0, reported=${gitea_repository.repotest0.username} ${gitea_repository.repotest0.name}, actual=${gitea_repository.repotest0.clone_url}"
}
#
# poorly-controlled dependency, indeterminate result
# result matches declaration after at most 2 apply operations
#
resource "gitea_org" "repotest1" {
  name = "RepoTest1"
}
resource "gitea_repository" "repotest1" {
  username = "RepoTest1"
  name         = "repotest1"
}
output repotest1 {
  value = "declared=RepoTest1 repotest1, reported=${gitea_repository.repotest1.username} ${gitea_repository.repotest1.name}, actual=${gitea_repository.repotest1.clone_url}"
}
#
# repo in non-existent org, undesirable result
# result never matches declaration regardless of how many apply operations
#
resource "gitea_repository" "repotest2" {
  username = "RepoTest2"
  name         = "repotest2"
}
output repotest2 {
  value = "declared=RepoTest2 repotest2, reported=${gitea_repository.repotest2.username} ${gitea_repository.repotest2.name}, actual=${gitea_repository.repotest2.clone_url}"
}
```
Apply this on a pre-existing gitea instance. Result is
```
(dev) juddw@N-20MAPF1PLWX7:~/trash/terraform-provider-gitea/repotest$ terraform apply --auto-approve -var 'gitea_url=<myGiteaUrl>' -var 'gitea_token=<myGiteaToken>'
gitea_org.repotest0: Creating...
gitea_org.repotest1: Creating...
gitea_repository.repotest2: Creating...
gitea_repository.repotest1: Creating...
gitea_org.repotest0: Creation complete after 0s [id=18]
gitea_org.repotest1: Creation complete after 0s [id=19]
gitea_repository.repotest0: Creating...
gitea_repository.repotest1: Creation complete after 2s [id=57]
gitea_repository.repotest0: Creation complete after 2s [id=59]
gitea_repository.repotest2: Creation complete after 3s [id=58]

Apply complete! Resources: 5 added, 0 changed, 0 destroyed.

Outputs:

repotest0 = "declared=RepoTest0 repotest0, reported=RepoTest0 repotest0, actual=https://gitea.jowtest-krhmq-hc282.578356578901.aws.nokia-solution.com/RepoTest0/repotest0.git"
repotest1 = "declared=RepoTest1 repotest1, reported=gitea_admin repotest1, actual=https://gitea.jowtest-krhmq-hc282.578356578901.aws.nokia-solution.com/gitea_admin/repotest1.git"
repotest2 = "declared=RepoTest2 repotest2, reported=gitea_admin repotest2, actual=https://gitea.jowtest-krhmq-hc282.578356578901.aws.nokia-solution.com/gitea_admin/repotest2.git"
```

**Expected behavior**
```
repotest0 = "declared=RepoTest0 repotest0, reported=RepoTest0 repotest0, actual=https://gitea.jowtest-krhmq-hc282.578356578901.aws.nokia-solution.com/RepoTest0/repotest0.git"
repotest1 = expect apply to FAIL when RepoTest1 org does not (yet) exist
repotest2 = expect apply to FAIL because RepoTest2 org noes not exist
```

**Log Output**
If applicable, add logs to help explain your problem.

**Additional Data**
Important for reproducability.

- Terraform Version
0.14.9, but I don't think it matters

- Operating System
Ubuntu 22.04 on WSL on Win10

- Provider Version
0.11.0

**Comment by Lerentis** [comment]
Thanks for reporting this @juddowilcox . 
I can not remember [why i did it like this](https://github.com/Lerentis/terraform-provider-gitea/blob/main/gitea/resource_gitea_repository.go#L147) but i agree that this is unexpected behaviour. I will work on a fix for this probably during the weekend

**Comment by Lerentis** [comment]
Fixed via [v0.11.1 ](https://github.com/Lerentis/terraform-provider-gitea/releases/tag/v0.11.1)
